Resources/public/js/apps/ez-platformuiapp.js

```diff
@@ -140,8 +142,20 @@ YUI.add('ez-platformuiapp', function (Y) {
 
             this.on({
                 'loadingChange': this._loading,
-                'navigate': function () {
+                '*:contentDiscover': function() { this.set('isDiscovering', true)},
+                '*:cancelDiscover': function() { this.set('isDiscovering', false)},
+                '*:contentDiscovered': function() { this.set('isDiscovering', false)},
+                '*:navigate': function () {
                     this.set('loading', true);
+                },
+
+            });
+
+            Y.on('hashchange', function (e) {
+                if ( app.get('isDiscovering') && e.newHash != app.get('hash') ) {
```

Comment:
this is for closing the UDW ( by firing cancelDiscover ) but mostly to keep the same hash (which is part of the bug and not for the improvement)